context/context.go

```diff
@@ -13,12 +12,14 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+type Context context.Context
```

Comment:
It's not immediately obvious to me what this does. Can you show an example in the description of the PR?

Reply:
Added an example

Reply:
I'm essentially subclassing the context package with our own helpers.